README
NAME
    Pinto::Common - Common libraries for the Pinto suite

VERSION
    version 0.070

DESCRIPTION
    This distribution exists to provide modules that are used across pinto
    and pintod. There are no user-serviceable parts in this distribution.

SUPPORT
  Perldoc
    You can find documentation for this module with the perldoc command.

      perldoc Pinto::Common

  Websites
    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

    *   Search CPAN

        The default CPAN search engine, useful to view POD in HTML format.

        <http://search.cpan.org/dist/Pinto-Common>

    *   CPAN Ratings

        The CPAN Ratings is a website that allows community ratings and
        reviews of Perl modules.

        <http://cpanratings.perl.org/d/Pinto-Common>

    *   CPAN Testers

        The CPAN Testers is a network of smokers who run automated tests on
        uploaded CPAN distributions.

        <http://www.cpantesters.org/distro/P/Pinto-Common>

    *   CPAN Testers Matrix

        The CPAN Testers Matrix is a website that provides a visual overview
        of the test results for a distribution on various Perls/platforms.

        <http://matrix.cpantesters.org/?dist=Pinto-Common>

    *   CPAN Testers Dependencies

        The CPAN Testers Dependencies is a website that shows a chart of the
        test results of all dependencies for a distribution.

        <http://deps.cpantesters.org/?module=Pinto::Common>

  Bugs / Feature Requests
    <https://github.com/thaljef/Pinto-Common/issues>

  Source Code
    <https://github.com/thaljef/Pinto-Common>

      git clone git://github.com/thaljef/Pinto-Common.git

AUTHOR
    Jeffrey Ryan Thalhammer <jeff@stratopan.com>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2011 by Imaginative Software Systems.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
0.070     2013-04-06 12:17:35 America/Los_Angeles

    * Prevent 'uninitialized' warning seen on perl 5.14.  Thanks @akarelas!
    
0.069     2013-04-03 21:42:32 America/Los_Angeles

    * No code changes.  Just revised documentation.

0.068     2013-03-29 23:36:45 America/Los_Angeles

    * No code changes.
    
    * Revised Pinto::Manual::Installing.

0.067     2013-03-28 20:26:28 America/Los_Angeles

    * No code changes.

    * Now requires Module::Corelist 2.85 or later, so you can target perls
    even newer than the one you have installed now.

    * Thank you CPAN Testers!

0.066     2013-03-25 21:06:00 America/Los_Angeles

    * Added LOCK_TYPE constants.

    * Added PkgSpecList and DistSpecList types.

    * Added Username and RevisionID types.

    * Specs type has been renamed to SpecList.

    * Files type has been renamed to FileList.

    * Author type has been renamed to AuthorID

    * Added several new utility functions used by Pinto.

    * PackageSpec now has is_core() and is_perl() convenience methods.

    * Rewrote functions for handling errors & warnings.

0.065_04  2013-03-20 22:50:55 America/Los_Angeles
0.065_03  2013-03-20 15:44:42 America/Los_Angeles
0.065_02  2013-03-19 15:06:03 America/Los_Angeles
0.065_01  2013-03-15 16:06:23 America/Los_Angeles

    * Changes consolidate above under release 0.066

0.064     2012-11-14 00:22:32 America/Los_Angeles

     * Added a couple utility functions I need in Pinto.

0.063     2012-11-08 09:32:10 America/Los_Angeles

     * Added StackObject type.

     * Fixed typo that resulted in syntax error (thanks Schwern).

0.062     2012-11-07 01:21:34 America/Los_Angeles

     * This distribution now contains Pinto::Util, which
     used to be part of the Pinto core.  To make PAUSE
     happy, it needs to have a higher version number than
     the last release when it was in the Pinto core. Hence
     the big jump in the version number since the last
     release of this distribution.

0.050     2012-10-22 12:34:27 America/Los_Angeles

     * Revised documentation.

0.049     2012-10-20 00:02:17 America/Los_Angeles

     * Author, StackName, and PropertyName types are no longer
     to be in all lower or upper case. 

0.048     2012-09-20 12:21:53 America/Los_Angeles

     * Revised and expanded documentation to cover new commands.

0.047     2012-09-17 13:07:15 America/Los_Angeles

     * Prevent 'uninitialized' warnings in type constraints.

0.046     2012-08-21 21:22:22 America/Los_Angeles

     * Just documentation edits.

0.045     2012-07-15 01:21:02 America/Los_Angeles

     * Added StackAt and StackDefault types.

0.044     2012-05-21 23:28:19 America/Los_Angeles

     * Moved Pinto::Role::PauseConfig over from Pinto-Core.
     This module is also needed by the D::Z plugin.

0.043     2012-05-17 22:12:42 America/Los_Angeles

     * Added a Pinto::Manual::Introduction to explain
     some whys and wheres.

0.042     2012-05-15 11:43:35 America/Los_Angeles

     * Added a main Pinto::Manual page, that redirects you
     to Pinto::Manual::Tutorial or Pinto::Manual::QuickStart

0.041     2012-05-15 11:07:07 America/Los_Angeles

     * Removed unused roles and classes.

     * Added StackName and PropertyName types.

0.038     2012-04-24 22:02:17 America/Los_Angeles

     Bug Fixes:

     Fixed the 02-authorable test, which would fail if you had a
     ~/.pauserc file.

0.037     2012-04-16 18:11:52 America/Los_Angeles

    Added DistributionSpec and PackageSpec classes.

    Enhanced the Import action to accept either a DistributionSpec or
    PackageSpec as the target.

0.036     2012-04-08 23:38:19 America/Los_Angeles

    Reorganized roles into different directory/package structure.

0.035     2012-04-04 18:55:44 America/Los_Angeles

    Revised the Pinto::Manual.  This is still a work in progress.

    This distribution now contains several Roles and other
    bits that are shared by both Pinto and Pinto::Remote.

0.020     2011-12-02 03:01:01 America/Los_Angeles

    Renamed the URI type to Uri, to avoid conflict
    with the URI class name.

0.019     2011-11-23 15:36:33 America/Los_Angeles

    Added the Vers type (for version objects).

0.018     2011-08-30 01:13:35 America/Los_Angeles

    Added Pinto::Constants, for constant values that are shared
    by the various pinto-* utilities.

0.017     2011-08-24 00:20:47 America/Los_Angeles

    The Logger and Config classes have been moved back into the
    Pinto distribution.  Now that they have a more specialized
    purpose, is no longer necessary to put them in a separate
    distribution.  So this dist now only contains Pinto::Types
    and Pinto::Manual.

0.015     2011-08-18 10:30:01 America/Los_Angeles

    Logger output is now colorized.  You can disable colorization in
    your configuration.

0.014     2011-08-17 22:56:04 America/Los_Angeles

    You can now tell Pinto::Logger where to write to.  This can be a
    file, a filehandle, or a scalar ref.  This is mostly useful for
    testing purposes.

0.013     2011-08-17 15:55:53 America/Los_Angeles

    Initial release.


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Common libraries for the L<Pinto> suite",
  "dist_author" => [
    "Jeffrey Ryan Thalhammer <jeff\@stratopan.com>"
  ],
  "dist_name" => "Pinto-Common",
  "dist_version" => "0.070",
  "license" => "perl",
  "module_name" => "Pinto::Common",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Class::Load" => 0,
    "DateTime" => 0,
    "Digest::MD5" => 0,
    "Digest::SHA" => 0,
    "English" => 0,
    "Exporter" => 0,
    "File::HomeDir" => 0,
    "IO::File" => 0,
    "IO::Handle" => 0,
    "IO::Interactive" => 0,
    "IO::String" => 0,
    "Module::CoreList" => "2.85",
    "Moose" => 0,
    "Moose::Role" => 0,
    "MooseX::MarkAsMethods" => 0,
    "MooseX::Types" => 0,
    "MooseX::Types::Moose" => 0,
    "Path::Class" => 0,
    "Path::Class::Dir" => 0,
    "Path::Class::File" => 0,
    "Readonly" => 0,
    "Scalar::Util" => 0,
    "Throwable::Error" => 0,
    "URI" => 0,
    "UUID::Tiny" => 0,
    "base" => 0,
    "namespace::autoclean" => 0,
    "overload" => 0,
    "perl" => "5.008",
    "strict" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "FindBin" => 0,
    "Test::Exception" => 0,
    "Test::More" => 0,
    "Test::Warn" => 0
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
